server/actions/waivers.js

```diff
@@ -1,10 +1,52 @@
-import { unlink } from "fs/promises";
-
-export const deleteWaiver = async (id) => {
-  try {
-    await unlink(`public/files/${id}.pdf`);
-  } catch (e) {
-    return { status: 400, message: "Delete file operation failed" };
-  }
-  return { status: 200, message: "File deleted successfully" };
-};
+const WaiverData = require("../mongodb/models/waivers");
+import dbConnect from "../mongodb/index";
+const User = require("../mongodb/models/user");
+const mongoose = require("mongoose");
+
+export async function createWaiver(type, text, organizationId, next) {
+  const newWaiver = new WaiverData(type, text, organizationId);
+
+  await dbConnect();
+
+  const waiver = await WaiverData.create({
+    type: type,
+    text: text,
+    organizationId: organizationId,
+  });
+}
+
+export async function getWaiver(type, organizationId) {
+  await dbConnect();
+
+  return WaiverData.find({ type: type, organizationId }).then((waiver) => {
+    return waiver;
+  });
+}
+
+export async function updateWaiver(waiverData, organizationId, next) {
+  await dbConnect();
+
+  return WaiverData.findOneAndUpdate(
```

Comment:
You should combine the updateWaiver and createWaiver function by using upsert: true in the `findOneAndUpdate`. Essentially, if it doesn't find a document that matches the filter (type + orgId), it will create it. Then on the API side, you can just have one endpoint to update the waiver (which will either create the waiver or update the text).